src/main/java/io/fabric8/maven/docker/access/log/LogRequestor.java

```diff
@@ -91,6 +91,8 @@ public void fetchLogs() {
             parseResponse(resp);
         } catch (IOException exp) {
             callback.error(exp.getMessage());
+        } finally {
+            callback.close();
```

Comment:
We should ensure, that, when the callback is closed that this any call to the `LogRequestors` method is invalid (by throwing an exception).

Its also not so nice that the responsibilities for creating the callback (== caller of constructor of `LogRequestor`) and the one closing the callback (== the `LogRequestor`) are different objects. It would be much better when the `LogRequestor` has the complete control over allocation and closing of the print stream.

I suggest therefor to introduce a an `LogCallback open()` which allocates the stream instead of automatically allocating it in the constructor. Each method which can close a stream then would have to open it before.